crates/orchestrator/src/shared/types.rs

```diff
@@ -55,3 +58,39 @@ pub struct ChainDefaultContext<'a> {
     pub default_db_snapshot: Option<&'a AssetLocation>,
     pub default_args: Vec<&'a Arg>,
 }
+
+#[derive(Debug, Clone)]
+pub struct RegisterParachainOptions {
+    pub para_id: u32,
+    pub wasm_path: PathBuf,
+    pub state_path: PathBuf,
+    pub node_ws_url: String,
+    pub onboard_as_para: bool,
+    pub seed: Option<[u8; 32]>,
+    pub finalization: bool,
+}
+
+#[derive(Debug, Clone)]
+pub struct ParachainGenesisArgs {
+    pub genesis_head: String,
+    pub validation_code: String,
+    pub parachain: bool,
+}
+
+impl<T> AsRef<T> for ParachainGenesisArgs
+where
+    T: ?Sized,
+    <ParachainGenesisArgs as Deref>::Target: AsRef<T>,
+{
+    fn as_ref(&self) -> &T {
+        self.deref().as_ref()
+    }
+}
+
+impl Deref for ParachainGenesisArgs {
+    type Target = String;
+
+    fn deref(&self) -> &Self::Target {
+        &self.genesis_head
+    }
+}
```

Comment:
Do we need this? why are you deref to the genesis_head?

Reply:
The `subxt::dynamic::tx` receives  the `call_data` as a `vec![]` of `Values::from_bytes`...
The call is:
```rs
let schedule_para = subxt::dynamic::tx(
            "ParasSudoWrapperCall",
            "sudo_schedule_para_initialize",
            vec![
                Value::from_bytes(account_id),
                Value::from_bytes(parachain_genesis_value), // <--- custom type `ParachainGenesisArgs`
            ],
        );
```

In order to satisfy the the trait bound `ParachainGenesisArgs: AsRef<[u8]>`  needs to be implemeted but I am not sure how to Deref for the whole struct. 
For a temp soluton I left the string of `genesis_head` - but a correct struct should return